src/drive/web/modules/navigation/RealtimeFiles.jsx

```diff
@@ -93,7 +93,9 @@ export class RealtimeFiles extends React.Component {
   }
 
   onDocumentDeletion = doc => {
-    if (this.isInCurrentView(doc)) this.props.deleteFile(this.normalizeId(doc))
+    //Ne pas faire un deleteFile quand on vide la corbeille car on a déjà une action pour cela
```

Comment:
:fr:

Reply:
See : 
 `
I had to hack this realtime line (4eeea76#diff-9d445e7faef1808a83ef82ea6ba523efR98) because we're dispatching the exact same action for a deletion and to empty the trash.
Quand dans un nfolder, on supprimer des fichiers : 
- TRash_files(ids)
 + delete_file(id) (x X)


Quand on supprime depuis la corbeille des fichiers définitivement : 
- Destroy_files(ids)
    + delete_file(id) (x X)
But the delete_file is not dispatched by the click, but by the real time (4eeea76#diff-9d445e7faef1808a83ef82ea6ba523efR99). Since we're removing files here, we can have a race between this and this line

cozy-drive/src/drive/web/modules/trash/actions.js

Line 79 in 4eeea76

 ids: files.map(f => f.id), 
. Sometimes, the map doesn't have the right value.

`
https://github.com/cozy/cozy-drive/pull/1611#issuecomment-485674260